js/sparkswap.js

```diff
@@ -233,8 +232,24 @@ module.exports = class sparkswap extends Exchange {
         return this.privateGetV1OrderId ({ id });
     }
 
-    async fetchOrderBook () {
-        throw new ExchangeError ('Not Implemented');
+    async fetchOrderBook (market, params = {}) {
+        await this.loadMarkets ();
+        // This call will return the correct information that CCXT will expect, however
+        // we need to modify the payload to convert nanosecond timestamps to miliseconds
+        //
+        const res = await this.privateGetV1Orderbook ({ 'market': this.marketId (market) });
+        // We remove the nanoseconds from the millisecond timestamp to fit the CCXT
+        // format
+        const millisecondTimestamp = res.timestamp.substring (0, res.timestamp.length - 6);
+        // We remove the nanoseconds (4 characters) from the ISO nanosecond timestamp to fit
+        // the ccxt format
+        const millisecondDatetime = res.datetime.substring (0, res.datetime.length - 5) + 'Z';
+        return {
+            'asks': res.asks,
```

Comment:
Have you checked that bids and asks are formatted correctly? I think they are but just checking

Reply:
Yup. we are all good.
<img width="394" alt="screen shot 2018-10-03 at 4 06 42 pm" src="https://user-images.githubusercontent.com/5478311/46444364-79be4f00-c726-11e8-90c7-828197f04875.png">